cluster/watchdog.go

```diff
@@ -121,6 +121,9 @@ func (w *Watchdog) rescheduleContainers(e *Engine) {
 			}
 		}
 
+		// Clear out the network configs, since we cannot create a
+		// container with multiple networks.
+		c.Config.NetworkingConfig.EndpointsConfig = nil
```

Comment:
One of the unfortunate things about this piece of code is that some information, particularly endpoints are replicated in multiple locations. Make doubly sure that this works on testing. ie `c.Config.NetworkingConfig.EndpointsConfig` is the data structure you want to change.